packages/flutter/lib/src/widgets/display_feature_sub_screen.dart

```diff
@@ -0,0 +1,157 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show DisplayFeature;
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/rendering.dart';
+import 'package:vector_math/vector_math_64.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'media_query.dart';
+import 'overlay.dart';
+
+/// Positions [child] such that it avoids overlapping any [DisplayFeature] that
+/// splits the screen into sub-screens.
+///
+/// After determining the sub-screens, the closest one to [anchorPoint] is used
+/// render the [child].
+///
+/// If no [anchorPoint] is provided, then [Directionality] is used. If no
+/// [Directionality] and no [anchorPoint] are provided, then the top-left screen
+/// is used.
+///
+/// See also:
+///
+///  * [showDialog], which is a way to display a DialogRoute.
+///  * [showCupertinoDialog], which displays an iOS-style dialog.
+class DisplayFeatureSubScreen extends StatelessWidget {
+  /// Creates a widget that positions its child so that it avoids display features.
+  const DisplayFeatureSubScreen({
+    Key? key,
+    required this.child,
+    this.anchorPoint,
+  }) : super(key: key);
+
+  /// The child that will avoid the display features.
+  final Widget child;
+
+  /// The anchor point used to pick the closest area that has no display features.
+  /// `Offset(0,0)` is the top-left corner of the available screen space. For
```

Comment:
nit: first paragraph should be a single sentence. Add an empty line before this one?